lib/ansible/playbook/task.py

```diff
@@ -455,6 +455,15 @@ def set_loader(self, loader):
         if self._parent:
             self._parent.set_loader(loader)
 
+    def get_parent_block_name(self):
+
+        name = ''
+        parent_block = self.get_first_parent_block(static=True)
+        if parent_block:
+            name = getattr(parent_block, 'name', '')
+
+        return name
```

Comment:
Since the `getattr()` call has a fallback, I don't think that conditional is necessary:
```suggestion
        parent_block = self.get_first_parent_block(static=True)
        return getattr(parent_block, 'name', '')
```
Am I missing anything?

Reply:
nope, i used to have more code there, but kept if after removing